File: backend/models/note.go
package models

import "gorm.io/gorm"

type Note struct {
	gorm.Model
	Title       string `gorm:"size:255;not null;unique" json:"title"`
	Description string `gorm:"size:255;not null;" json:"description"`
	Status      string `gorm:"size:255;not null;" json:"status"`
	// UserId      string `gorm:"size:255;not null;unique" json:"userId"`
}

File: frontend/src/app/dto/note.ts
export class Note {
    noteId: string;
    userId: string;
    title: string;
    description: string;

    constructor(noteId: string, userId: string, title: string, description: string) {
        this.noteId = noteId;
        this.userId = userId;
        this.title = title;
        this.description = description;
    }
}
